gum/backend-linux/gumprocess-linux.c

```diff
@@ -551,11 +552,20 @@ gum_do_modify_thread (gpointer data)
     goto failed_to_attach;
   attached = TRUE;
 
-  wait_result = gum_libc_waitpid (ctx->thread_id, &status, 0);
-  if (wait_result != ctx->thread_id || !WIFSTOPPED (status) ||
-      WSTOPSIG (status) != SIGSTOP)
+  wait_result = gum_libc_waitpid (ctx->thread_id, &status, __WALL);
```

Comment:
```suggestion
  wait_result = gum_libc_waitpid (ctx->thread_id, &status, __WALL);
```

When is `__WALL` needed here?

Reply:
The man page states...

```
__WCLONE
    Wait for "clone" children only. If omitted then wait for "non-clone" children only. (A "clone" child is one which
    delivers no signal, or a signal other than SIGCHLD to its parent upon termination.) This option is ignored if __WALL
    is also specified. 
__WALL (since Linux 2.4)
    Wait for all children, regardless of type ("clone" or "non-clone"). 
```

Experimentation has shown that on old 3.x kernels, the wait is not satisfied without `__WALL` being specified. This flag will mainly effect behaviour in the instance that the PID is specified to wait for multiple threads (e.g. <-1, -1, or 0), but since we are waiting for a specified thread, it cannot have any negative consequences.